.github/workflows/release-opampsupervisor.yaml

```diff
@@ -22,23 +22,51 @@ jobs:
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           fetch-depth: 0
+
+      - name: Set goreleaser last tag reference in case of non-nightly release
+        id: prev-tag
+        if: ${{ !contains(github.ref, '-nightly') }}
+        # find previous tag by taking only OpAMP supervisor tags, filtering out nightly tags and then choosing the
+        # second to last tag (last one is the current release)
+        run: |
+          prev_tag=$(git tag | grep "cmd/opampsupervisor" | grep -v "nightly" | sort -r --version-sort | head -n 2 | tail -n 1)
+          echo "PREVIOUS_RELEASE_TAG=$prev_tag" >> "$GITHUB_OUTPUT"
+
+      - name: Set nightly enabled
+        id: nightly-check
+        if: ${{ contains(github.ref, '-nightly') }}
+        run: |
+          echo "NIGHTLY_FLAG=--nightly" >> "$GITHUB_OUTPUT"
+
+      - name: Set COLLECTOR_REF
+        id: collector-ref
+        run: |
+          if [ '${{ contains(github.ref, '-nightly') }}' == 'true' ]; then
+            echo "COLLECTOR_REF=main" >> "$GITHUB_OUTPUT"
+          else
+            echo "COLLECTOR_TAG=${{ github.ref }}" >> "$GITHUB_OUTPUT"
+          fi
```

Comment:
I noticed that these 3 steps are repeated in a few places. Could we somehow extract them into something reusable? Not required for now, but in the near term future it would be useful so that fixes and changes could easily be applied in all places.